### DISCUSSION TITLE
Using Pyvista with pyside2 QT GUI

### DISCUSSION THREAD

**Discussion by ndahiya3** [pyvista/pyvista#2706] [Q&A]
I was wondering if it is possible to use PyVista inside pyside2 QT apps?

**Comment by akaszynski** [comment]
It is! Please see https://qtdocs.pyvista.org/

**Reply by ndahiya3** [reply]
I tried installing pyvistaqt with pyside2 in anaconda environment. The following code is throwing an error:

`import pyvista as pv
from pyvistaqt import BackgroundPlotter
plotter = BackgroundPlotter()`

vtkOpenGLRenderWindow.c:569    ERR| vtkEGLRenderWindow (0x55b64fde5af0): GLEW could not be initialized: Missing GL version
ERROR:root:GLEW could not be initialized: Missing GL version
Segmentation fault (core dumped)

Pyvista alone is working, I am able to run the "globe" demo.

**Reply by ndahiya3** [reply]
`conda install -c conda-forge pysideqt` seems to install `vtk 9.0.3 ` but `conda install -c conda-forge pyside2` upgrades vtk to `vtk 9.1.0` . Could that be the reason? Here's output of conda list:
[conda_list.txt](https://github.com/pyvista/pyvista/files/8793481/conda_list.txt)

**Comment by akaszynski** [comment]
Looks like there's an issue running pyvista with EGL. I have minimal experience with conda, this works locally:

```py
import sys

from qtpy import QtWidgets

import pyvistaqt
import pyvista as pv


if __name__ == '__main__':
    app = QtWidgets.QApplication(sys.argv)
    plotter = pyvistaqt.BackgroundPlotter()
    plotter.add_mesh(pv.Sphere(), smooth_shading=True)
    sys.exit(app.exec_())
```

With the following python packages from `pip freeze > requirements.txt`
[requirements.txt](https://github.com/pyvista/pyvista/files/8793798/requirements.txt)

Can you try creating a [venv](https://docs.python.org/3/library/venv.html) and install the above requirements with
```
pip install -r requirements.txt
```

**Reply by ndahiya3** [reply]
git@github.com: Permission denied (publickey).
  fatal: Could not read from remote repository.

  Please make sure you have the correct access rights
  and the repository exists.
ERROR: Command errored out with exit status 128: git clone -q 'ssh://****@github.com/pyvista/pyvistaqt.git' /some/path/pyvenv/src/pyvistaqt Check the logs for full command output.

**Reply by ndahiya3** [reply]
The local install according to your instructions works!
